### ISSUE TITLE
ProgramModuleObj.isCompleted() should take a user

### ISSUE THREAD

**Issue by benjaminjkraft** [learning-unlimited/ESP-Website#1968]
Right now `ProgramModuleObj.isCompleted()` just takes `self` as a parameter.  Nearly all callers of it already have the current user available, and nearly all implementations of it then end up calling `get_current_request().user` to get the current user.  This is ugly; we should pass the user as an argument.

**Comment by mgersh** [comment]
The reason why this isn't the case already is because `isCompleted()` is intended to be used in a template, so it can't take arguments other than `self`. But I completely agree that this is ugly and we should find a way around it.

**Comment by imran-techvoyager** [comment]
Hi @willgearty , I’ve been looking into this issue and took some time to trace how `isCompleted()` is currently used across module handlers.

One thing I noticed is that `getModules(user, …)` already attaches `self.user` to each module instance when a user is available. Some handlers (e.g. `availabilitymodule`) already support an optional `user` parameter and fall back to `self.user` before using `get_current_request()`, but others skip that step and go straight to `get_current_request()`.

A possible cleanup would be to:

* Standardize the base signature to `def isCompleted(self, user=None)`
* Prefer the explicit `user` argument
* Fall back to `self.user` (when set by `getModules`)
* Keep `get_current_request()` only as a final fallback for template usage

This should let us remove the inconsistency across handlers while remaining backward-compatible with existing template calls like `{{ mod.isCompleted }}`.

If this direction sounds reasonable, I’d be happy to take this on.

**Comment by willgearty** [comment]
Yep, that sounds like a good plan to me. Let me know if you have any questions as you move forward.
